server/index.js

```diff
@@ -1,24 +1,35 @@
 // server/index.js
 import cors from 'cors';
 import express from 'express';
+import 'dotenv/config'
 
 const app = express();
 app.use(cors());
 
 //Set the port that you want the server to run on
 const PORT = process.env.PORT || 8080;
+const APIKEY = process.env.API_KEY
+
 
 //creates an endpoint for the route /api
-app.get('/api', (req, res) => {
-  res.json({ message: 'Hello from ExpressJS' });
+app.get('/', (req, res) => {
+  res.send('This is the home page')
+})
+
+app.get('/api', async (req, res) => {
+  const url = `https://api.openweathermap.org/data/2.5/weather?q=${req.query.city}&appid=${APIKEY}`;
+  const response = await fetch(url)
+  const weatherData = await response.json()
+  res.json(weatherData);
+  console.log(weatherData)
```

Comment:
I know you're still working on this, but don't forget to remove unnecessary console.log()s when you are ready to submit your final version of this project.

Reply:
+1. For my comments below, no action required.

Sometimes it's useful for *documentation* to keep console logging, but to disable those logs when in production (e.g. https://stackoverflow.com/a/8002527/2562787).

For example, if you see the code (in production)

```
if (emailAddress.match(/^\S+@\S+$/g)) {
  ...
}
```

You may not know what it does. This is an example of non-obvious code deserving of a comment:

```
// Check if the email address is valid
if (emailAddress.match(/^\S+@\S+$/g)) {
  ...
}
```

You can make it one step better though, by replacing the comment with non-production logging:

```
console.log(`Checking if "${emailAddress}" is valid`);
if (emailAddress.match(/^\S+@\S+$/g)) {
  ...
}
```

This way has the benefit that if you're trying to debug a problem in the future, you don't have to go back and re-add logging, you only need to turn it on.

For this reason, most logging libraries in most languages have the concept of logging "levels", and the application can then be configured to log (or not log) at the various levels. Often the levels are named things like "debug", "info", "warn", and "error". For example, browsers have

* debug - https://developer.mozilla.org/en-US/docs/Web/API/console/debug
* info - https://developer.mozilla.org/en-US/docs/Web/API/console/info
* warn - https://developer.mozilla.org/en-US/docs/Web/API/console/warn

... however in the browser case you also need to consider whether you want a knowledgeable user to be able to enable your logging.

Reply:
@eatnumber1 In my past experience we've simply surrounded a console.xxx() with a condition checking if the environment is not production, but there's probably better ways to do it? I haven't thought too much about this so thanks for the illuminating comments about this! @jiajdo what he says is completely valid, but try to reduce the number of console.xxx() you submit to the shared set of code everyone in the project/company are using. Because if everyone starts logging a ton of stuff in the shared code, the console output in browser devtools will become cluttered. From my experience, if they do allow adding logging for non-prod (production) envs (environments), it's a temporary measure and later they are removed.

Reply:
Noted. Thanks for extra resources.